### PR TITLE
Feature/fix dns plugin logic

### PR DESCRIPTION
**Pull request by econchick**
**What this PR does / why we need it**:

This updates the logic of the Reconciler to take care of instantiating an auth + dns client that it needs to work with, making it easier for the janitor core to configure & run the reconciler as a plugin. Working on this made me realize that I've missed some required docs for this plugin (mainly configuration).

I've also renamed modules to make them more in line with @matt0bi 's naming scheme in [his branch](https://github.com/spotify/gordon-janitor-gcp/tree/add_gce_instance_authority).

**Which issue(s) this PR fixes**

N/A

**Special notes for your reviewer**:

There were many times that I stepped away from this patch, so I could have very well missed something/messed something up. I ask for reviewers' understanding :)

**Release note**:

```release-note
NONE
```

cc @spotify/alf

### CODE REVIEW

File: gordon_janitor_gcp/reconciler.py [hunk 1]
```diff
@@ -75,33 +72,63 @@ class GoogleDNSReconciler:
     state.
 
     Once validation is done, the Reconciler will emit a ``None`` message
-    to the ``changes_channel`` queue signalling to the ``Publisher``
-    (TODO: update name of publisher class once decided) that there are
-    no more change messages to expect.
+    to the ``changes_channel`` queue, signalling a Publisher client
+    (e.g. ``gpubsub_client.AIOGooglePubsubClient``) to publish the
+    message to a pub/sub to which the `Gordon
```
Comment (matt0bi) [inline gordon_janitor_gcp/reconciler.py]: I think the _the_ in _to which the `Gordon` subscribes_ can be omitted.

File: tests/unit/test_reconciler.py [hunk 2]
```diff
@@ -175,7 +212,7 @@ def test_reconciler_default(timeout, dns_client):
     # happy path
     [{'zone': 'example.net.', 'rrsets': []}, 1, 1],
     # no rrsets key
-    [{'zone': 'example.net.'}, 3, 0],
+    [{'zone': 'example.net.'},  3, 0],
```
Comment (matt0bi) [inline tests/unit/test_reconciler.py]: Stray space? (`[{'zone': 'example.net.'}, _3, 0]`)

**Review by matt0bi — COMMENTED** [review]
Looks good! Only found 2 typos.

**Comment by econchick** [comment]
@matt0bi thanks! I fixed those two - might re-review/+1'ing?

**Comment by matt0bi** [comment]
LGTM :+1:

**Pull request merged**